cpu/stm32/include/periph/cpu_i2c.h

```diff
@@ -54,23 +61,23 @@ extern "C" {
  * @{
  */
 #define HAVE_I2C_SPEED_T
+#if STM32_I2C_VARIANT == 1
+/* i2c_1.c */
 typedef enum {
-#if defined(CPU_FAM_STM32F1) || defined(CPU_FAM_STM32F2) || \
-    defined(CPU_FAM_STM32F4) || defined(CPU_FAM_STM32L1) || \
-    defined(CPU_FAM_STM32MP1)
```

Comment:
I am not familiar with this header. There seemed to be a distinction between the STM families that has been completely removed. How does this work now? Got the distinction just dropped? Was it not need in the first place?

Reply:
The one implementation (`i2c_1.c`) already supported arbitrary I2C speeds, the other (`i2c_2.c`) did not. This adds support for arbitrary I2C speeds for the other (`i2c_2.c`) as well. Hence, both `enum` values are just for backward compatibility / portability now.

Reply:
No :-/ `i2c_1.c` doesn't allow arbitrary clock speeds. I need to revert that.